### PR TITLE
Extended CRUD example

### PR DESCRIPTION
**Pull request by kybarg**
1. Not everything is covered by tests.
2. Still haven't implemented DELETE functionality.
3. This is early example, needs review

### CODE REVIEW

**Comment by diegohaz** [comment]
Great, @kybarg 

I'll take some time this weekend to review that.

**Comment by kybarg** [comment]
@diegohaz Thanks a lot. Also I need help with DELETE button container. If button is on the item page I want it to redirect to list page after successful deletion, and if click it on the list page - remove item, and append one more item to page in place of removed or just reload page.

**Comment by advance512** [comment]
@kybarg - any chance you could fix the tests?
This is great additional content, and I'd like to base the GraphQL-based example code on it.

File: package.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
     "lint": "eslint src src-clean .storybook",
     "dev": "concurrently --kill-others \"nodemon --watch src\" \"node webpack/webpack-dev-server\"",
     "storybook": "start-storybook -p 9001 -c .storybook",
-    "build:clean": "del \"dist/!(.git*|Procfile)**\"",
+    "build:clean": "del-cli \"dist/!(.git*|Procfile)**\"",
```
Comment (diegohaz) [inline package.json]: Does this work? Is there some problem with `del`?

File: src/containers/PostForm.js [hunk 2]
```diff
@@ -1,35 +1,72 @@
...
+const mapStateToProps = (state, { id }) => {
+  if (id) {
+    return ({
+      initialValues: merge(
+        fromEntities.getDetail(state, 'post', id),
+        {
+          _csrf: fromForm.getCsrfToken(state)
+        }
+      )
+    })
```
Comment (diegohaz) [inline src/containers/PostForm.js]: I wouldn't use `lodash` here, but [`object spread`](https://github.com/sebmarkbage/ecmascript-rest-spread):
```js
return {
  initialValues: {
    _csrf: fromForm.getCsrfToken(state),
    ...fromEntities.getDetail(state, 'post', id)
  }
}
```

Also, you don't need to wrap with `()`

File: package.json (hunk 1 above)
Comment (kybarg) [inline package.json]: `del` is reserved command in WIndows. `del-cli` is just alias for `del` package. Does it work on Mac and Linux?

File: src/containers/PostForm.js (hunk 2 above)
Comment (kybarg) [inline src/containers/PostForm.js]: That looks much better)

**Comment by kybarg** [comment]
@advance512 This PR is in discussion stage. Will make tests later.

**Comment by kybarg** [comment]
@diegohaz I tried to use naming that could be simple to understand CRUD functionality. Can use other names if you insist 😄

File: package.json (hunk 1 above)
Comment (diegohaz) [inline package.json]: Oh, I see. Great. That's the problem on #65

**Comment by diegohaz** [comment]
> Also could you point me please how better to implement deleting functionality?

I'm working on another project that can help you: https://github.com/diegohaz/redux-modules

Take a look at the `resource` store.

It's WIP, but I'll basically explain there what I've been doing with redux stuff on ARc with some other examples (like `resource`).

**Comment by kybarg** [comment]
@diegohaz I added delete functionality to store. I need help with "delete button" 😄

**Comment by diegohaz** [comment]
>  If button is on the item page I want it to redirect to list page after successful deletion, and if click it on the list page

You can use the [`push` ](https://github.com/reactjs/react-router-redux#pushlocation-replacelocation-gonumber-goback-goforward) action creator from `react-router-redux` to redirect the user. As this is a side effect of the successful deletion, I would do that on the worker saga responsible for deleting the post (`deletePost`).

> and if click it on the list page - remove item, and append one more item to page in place of removed or just reload page.

I would also do that on the worker saga. Just verify the url, send a flag from the component to the deletion action creator or use another action creator for that page (as you have more control over the route state inside the components).

**Pull request closed without merging**